### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'bool' and 'str'

### ISSUE THREAD

**Issue by bradleybuilt** [frescobaldi/frescobaldi#1386]
I am brand new to Frescobaldi. I downloaded to MacOS 11.5.2. I tried to run my first file containing a sample from the documentation and got this error (attached below). Can you help me resolve this? I would love to be able to use the program.

<img width="712" alt="Screen Shot 2021-09-01 at 1 28 36 PM" src="https://user-images.githubusercontent.com/60622314/131724435-cfc2db66-7652-45e6-a83f-2699aeeb1dfd.png">

**Comment by fedelibre** [comment]
Can you attach the file here? I think you'll have to change the extension to .txt to pass Github filters.
I'll see if I can reproduce it. But it may be depend on your installation.

**Comment by bradleybuilt** [comment]
It is saved as a `.ly` file, but I cannot upload that file type here. Here is the code that generated the error (taken straight from the user guide):

```
\relative c'' {
  \time 7/4
  c2 bes4 a2 g a bes4 a( g) f2
}
\addlyrics {
  Join us now and share the soft -- ware!
}
```

**Comment by fedelibre** [comment]
Ok, so it's a very simple snippet. Of course it compiles fine on my 
Linux installation.
It seems definitely an issue with the Mac package.

**Comment by bradleybuilt** [comment]
Thank you. Do you think you can sort it out?

**Comment by fedelibre** [comment]
Maybe @dliessi can help.

It seems that Frescobaldi mac package has python 3.9.
My Linux installation is using python 3.8.8. I should test it with python 3.9 and see if I can reproduce it.

**Comment by stuartsierra** [comment]
I can report what appears to be the same issue.

On macOS Big Sur 11.6, Intel Core i7.
Frescobaldi installed from Homebrew cask (`brew install frescobaldi`).

1. Launch Frescobaldi
2. Select menus File -> New -> Blank Music Sheet
"Internal Error" dialog box appears:

```
Frescobaldi: 3.1.3
Extension API: 0.9.0
Python: 3.9.1
python-ly: 0.9.7
Qt: 5.11.3
PyQt: 5.15.2
sip: 4.19.24
poppler: 20.12.1
python-poppler-qt: 0.75.0
OS: macOS-10.16-x86_64-i386-64bit
installation kind: standalone .app bundle

Traceback (most recent call last):
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/snippet/menu.py", line 103, in slotTriggered
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/snippet/menu.py", line 167, in applySnippet
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 200, in engrave
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 252, in runJob
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/manager.py", line 61, in start_job
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/__init__.py", line 211, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/lilypond.py", line 176, in configure_command
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 138, in wait
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 261, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 278, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 189, in set
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 191, in emit
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 308, in call
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 276, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 306, in frommacports
TypeError: unsupported operand type(s) for +: 'bool' and 'str'
```

Similarly:

1. Launch Frescobaldi
2. Open any score file
3. Select menus Lilypond -> Engrave (preview)
"Internal Error" dialog box appears:

```
Traceback (most recent call last):
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 141, in engraveRunner
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 145, in engravePreview
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 200, in engrave
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/engrave/__init__.py", line 252, in runJob
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/manager.py", line 61, in start_job
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/__init__.py", line 211, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/job/lilypond.py", line 176, in configure_command
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 138, in wait
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 261, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 253, in start
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 278, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 189, in set
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 191, in emit
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/signals.py", line 308, in call
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 264, in checkstart
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/cachedproperty.py", line 276, in run
  File "/Applications/Frescobaldi.app/Contents/Resources/lib/python3.9/frescobaldi_app/lilypondinfo.py", line 306, in frommacports
TypeError: unsupported operand type(s) for +: 'bool' and 'str'
```

**Comment by smiledra** [comment]
I have encountered a problem that hopefully is similar to this (see below). My system is linux (archlinux) using Python 3.9. 

```
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/frescobaldi_app/engrave/__init__.py", line 149, in engravePublish
    self.engrave('publish')
  File "/usr/lib/python3.9/site-packages/frescobaldi_app/engrave/__init__.py", line 200, in engrave
    self.runJob(job_class(doc, args), doc)
  File "/usr/lib/python3.9/site-packages/frescobaldi_app/engrave/__init__.py", line 252, in runJob
    job.manager.manager(document).start_job(j)
  File "/usr/lib/python3.9/site-packages/frescobaldi_app/job/manager.py", line 61, in start_job
    job.start()
  File "/usr/lib/python3.9/site-packages/frescobaldi_app/job/__init__.py", line 225, in start
    self._update_process_environment()
  File "/usr/lib/python3.9/site-packages/frescobaldi_app/job/__init__.py", line 304, in _update_process_environment
    se.remove(k) if v is None else se.insert(k, v)
TypeError: arguments did not match any overloaded call:
  insert(self, str, str): argument 2 has unexpected type 'bool'
  insert(self, QProcessEnvironment): argument 1 has unexpected type 'str'
```

**Comment by dliessi** [comment]
@bradleybuilt, @stuartsierra:
This was reported again as #1406 and solved in master as #1397.
If we release 3.1.4 before 3.2 we will backport the fix.
In the meantime, as a workaround you can explicitly set the full path to LilyPond in Frescobaldi's preferences and patch one file to solve a problem with Ghostscript (#1356), see https://github.com/frescobaldi/frescobaldi/issues/1406#issuecomment-997359090.

@smiledra: Your problem is different.
I see that it happened to you with "Engrave (publish)", but I cannot reproduce it and I don't understand why it would happen.
I see also that the package definition on Arch has been recently updated.
I'm closing this issue, since the original problem is fixed.
Please test the new version of the Arch package and if the problem persists open a new issue for your problem.
